src/Illuminate/Database/Eloquent/HasBuilder.php

```diff
@@ -111,6 +121,24 @@ public static function onWriteConnection()
         return parent::onWriteConnection();
     }
 
+    /**
+     * Resolve the builder class name from the QueriedBy attribute.
+     *
+     * @return class-string<TBuilder>|null
+     */
+    public function resolveBuilderFromAttribute()
+    {
+        $reflectionClass = new ReflectionClass(static::class);
+
+        $attributes = $reflectionClass->getAttributes(QueriedBy::class);
+
+        if (! isset($attributes[0]) || ! isset($attributes[0]->getArguments()[0])) {
+            return;
+        }
+
+        return $attributes[0]->getArguments()[0];
```

Comment:
Instead of looking at the arguments (and checking if they are set), you can just directly get the attribute instance and access the property:


```
        $attribute = (new ReflectionClass(static::class))->getAttributes(QueriedBy::class)[0] ?? null;

        return $attribute ? $attribute->newInstance()->builderClass : null;
```